### ISSUE TITLE
Prioritized background activity, cause 10 appears after switch

### ISSUE THREAD

**Issue by SebastianWolf-SAP** [corona-warn-app/cwa-app-android#981]
## Avoid duplicates
* [x] Bug is not mentioned in the [FAQ](https://www.coronawarn.app/en/faq/)
* [x] Bug is specific for Android only, for general issues / questions that apply to iOS and Android please raise them in the [documentation repository](https://github.com/corona-warn-app/cwa-documentation)
* [x] Bug is not already reported in another issue

## Describe the bug

**Report from hotline:**
The user of the app clicks on settings and wants to enable the prioritized background activity.
Instead of the successful enabling of the feature the user receives the error message: "Ursache 10 irgendwas ist schiefgelaufen, diese Aktion ist leider aktuell nicht verfügbar. Bitte kontaktieren sie die technische Hotline" (engl: Cause 10 something went wrong, the action is currently not available. Please contact the technical hotline")

## Expected behaviour

See above

## Technical details

- Mobile device: Samsung A5 2015
- Android version:  6.0.1

**In case you see this behavior as well, please comment underneath mentioning your device and Android version. Thanks!**

---
Internal tracking ID: EXPOSUREAPP-2022

**Comment by m122-gh** [comment]
Hallo Herr Wolf,
Ich habe ein Samsung S5duos (G900FD) mit Android 6.0.1 und genau das beschriebene Problem.
Wie kann ich das beheben??

**Comment by PhilippNowak96** [comment]
This sounds like the problem described here: https://stackoverflow.com/questions/44862176/request-ignore-battery-optimizations-how-to-do-it-right/50840603#50840603. Unfortunately I have no Samsung running Android 6 to test this myself but maybe you guys can try to have a further look based on this information.

**Comment by vaubaehn** [comment]
> This sounds like the problem described here: https://stackoverflow.com/questions/44862176/request-ignore-battery-optimizations-how-to-do-it-right/50840603#50840603. Unfortunately I have no Samsung running Android 6 to test this myself but maybe you guys can try to have a further look based on this information.

@PhilippNowak96 looks like a good find!
Pushing forward to @svengabr

**Comment by svengabr** [comment]
When taking a look at the Jira tickets, this issue might be resolved with 1.3.0. Could you check again if the error still persists in the latest version?

EXPOSUREAPP-1885

Cheers

**Comment by vaubaehn** [comment]
hi @svengabr ,

> 
> When taking a look at the Jira tickets, this issue might be resolved with 1.3.0. Could you check again if the error still persists in the latest version?
> 
> EXPOSUREAPP-1885

just to prevent any confusion, and for double checking: this issue here is about CAUSE 10 (android.content.ActivityNotFoundException) - and obviously has more rare effects specifically on Samsung devices running Android 6. The link provided by @PhilippNowak96 https://stackoverflow.com/questions/44862176/request-ignore-battery-optimizations-how-to-do-it-right/50840603#50840603  seems to be rather useful. So, it's not about API 10...
Also, I couldn't find any related PR for 1.3.0 nor for 1.3.1 RC1.

Thanks for checking :)

**Comment by svengabr** [comment]
Hey @vaubaehn 

I just checked again today for the ticket I mentioned (EXPOSUREAPP-1885) and it was reopened yesterday by the development team. Internal the ticket was mentioned to be related this is why i thought it could already been fixed.

> Hi everyone, I have reopened this issue, since there is no background activity on the Xiaomi devices despite Google's watchdog as long as MIUI energymode is on (by default). BR Katharina

I just corrected the assignee here and gave it to @harambasicluka since he is working on the original Jira issue currently.

Best regards,
SG

Corona-Warn-App Open Source Team
